src/client/camera.cpp

```diff
@@ -695,16 +695,20 @@ void Camera::drawNametags()
 			i = m_nametags.begin();
 			i != m_nametags.end(); ++i) {
 		Nametag *nametag = *i;
-		v3f cam_pos = m_client->getEnv().getLocalPlayer()->getPosition() +
+		v3f cam_realpos = m_client->getEnv().getLocalPlayer()->getPosition() +
 				m_client->getEnv().getLocalPlayer()->getEyeOffset();
-		v3f nametag_pos = nametag->parent_node->getAbsolutePosition();
+		v3f nametag_realpos = nametag->parent_node->getAbsolutePosition() + intToFloat(m_client->getEnv().getCameraOffset(), BS);
 
+		cam_realpos     /= 10;
+		nametag_realpos /= 10;
+
+		errorstream << m_camera_position.X << ", " << m_camera_position.Y << ", " << m_camera_position.Z << std::endl;
 		if (!nametag->nametag_viewthrough_walls) {
-			if (cam_pos.getDistanceFrom(nametag_pos)/10 > stoi( // hack to get actual distance
+			if (cam_realpos.getDistanceFrom(nametag_realpos) > stoi(
 					g_settings->get("viewing_range")) + 16) // Add mapblock size in case pos is still loaded for client
```

Comment:
This comment seems unclear to me, why is `+ 16` required?

Also, settings can be directly obtained as numbers:
```suggestion
			// Add mapblock size in case pos is still loaded for client
			if (cam_realpos.getDistanceFrom(nametag_realpos) >
					(g_settings->getS16("viewing_range") + 16)
```

Reply:
16 is the mapblock size, changed that to use the macro.
The mapblocks don't disappear until all of their nodes are outside of the player's viewing range. And the entities/players won't always be at the edge of the mapblocks

Reply:
Ahh, I understand now. Thanks for clarifying.